### ISSUE TITLE
infer without results

### ISSUE THREAD

**Issue by wafaer** [wang-xinyu/tensorrtx#1411]
## Env

- GPU,3060
- OS, e.g. Ubuntu20.04
- Cuda version 11.4
- TensorRT version 8.2.4.2

## About this repo

- which branch/tag/commit are you using?
- which model? yolov5, retinaface?

## Your problem

I added an attention module to yolov5s-seg based on version 7.0 C3, trained 20 echos, and modified the model based on yours. After generating the engine, there was no result in reasoning. The code I added is as follows:

static ILayer* CoordAtt(INetworkDefinition *network, std::map<std::string, Weights>& weightMap, ITensor& input, int& inp, std::string lname)
{
    Dims dims = input.getDimensions();
    int h_size = dims.d[1];
    int w_size = dims.d[2];

    /// pool_h *************************************************************************************************************
    auto h_pool = network->addPoolingNd(input, PoolingType::kAVERAGE,DimsHW{1,1});
    h_pool->setStrideNd(DimsHW{1,1});
    h_pool->setPaddingNd(DimsHW{0,0});
    h_pool->setWindowSizeNd(DimsHW{1,w_size});
    ITensor* h_value = h_pool->getOutput(0);

    Permutation permutation = {1,0,2};
    nvinfer1::IShuffleLayer* shuffleLayer = network->addShuffle(*h_value);
    shuffleLayer->setFirstTranspose(Permutation(permutation));
    ITensor* perm1 = shuffleLayer->getOutput(0);

    /// pool_w *************************************************************************************************************
    auto w_pool = network->addPoolingNd(input, PoolingType::kAVERAGE,DimsHW{1,1});
    w_pool->setStrideNd(DimsHW{1,1});
    w_pool->setPaddingNd(DimsHW{0,0});
    w_pool->setWindowSizeNd(DimsHW{h_size,1});
    ITensor* w_value = w_pool->getOutput(0);

    Permutation permutation1 = {0,2,1};
    nvinfer1::IShuffleLayer* shuffleLayer1 = network->addShuffle(*w_value);
    shuffleLayer1->setFirstTranspose(Permutation(permutation1));
    ITensor* perm2 = shuffleLayer1->getOutput(0);

    Permutation permutation2 = {1,0,2};
    nvinfer1::IShuffleLayer* shuffleLayer2 = network->addShuffle(*perm2);
    shuffleLayer2->setFirstTranspose(Permutation(permutation2));
    ITensor* perm3 = shuffleLayer2->getOutput(0);

    /// torch.cat *************************************************************************************************************
    ITensor* sum_input_tensors[] = {perm1, perm3};
    ILayer* y = network->addConcatenation(sum_input_tensors,2);

    Permutation permutation3 = {1,0,2};
    nvinfer1::IShuffleLayer* shuffleLayer3 = network->addShuffle(*y->getOutput(0));
    shuffleLayer3->setFirstTranspose(Permutation(permutation3));
    ITensor* perm4 = shuffleLayer3->getOutput(0);

    int mip = MAX(8, inp/32);

    /// conv1 *************************************************************************************************************
    auto y_conv = convBlock2(network, weightMap, *perm4, mip, 1, 1, 1, lname + ".conv1",lname);

    /// torch.split *************************************************************************************************************
    Dims d = y_conv->getOutput(0)->getDimensions();
    ISliceLayer* sliceLayer1 = network->addSlice(*y_conv->getOutput(0), Dims3(0, 0, 0), Dims3(d.d[0], d.d[1] / 2, d.d[2]), Dims3(1, 1, 1));
    ISliceLayer* sliceLayer2 = network->addSlice(*y_conv->getOutput(0), Dims3(0, d.d[1] / 2, 0), Dims3(d.d[0], d.d[1] / 2, d.d[2]), Dims3(1, 1, 1));

    Permutation permutation4 = {0,2,1};
    nvinfer1::IShuffleLayer* shuffleLayer4 = network->addShuffle(*sliceLayer2->getOutput(0));
    shuffleLayer4->setFirstTranspose(Permutation(permutation4));
    ITensor* perm5 = shuffleLayer4->getOutput(0);

    /// conv_h conv_w *************************************************************************************************************
    auto a_h = convBlock3(network, weightMap, *sliceLayer1->getOutput(0), inp, 1, 1, 1, lname + ".conv_h");
    auto a_w = convBlock3(network, weightMap, *perm5, inp, 1, 1, 1, lname + ".conv_w");

    ITensor* sum_input_tensors1[] = {&input, a_h->getOutput(0),a_w->getOutput(0)};
    auto ew1 = network->addElementWise(*sum_input_tensors1[0], *sum_input_tensors1[1], ElementWiseOperation::kPROD);
    auto ew2 = network->addElementWise(*ew1->getOutput(0), *sum_input_tensors1[2], ElementWiseOperation::kPROD);

    return ew2;
}
What do you recommend?

**Comment by wang-xinyu** [comment]
@wafaer Hi, it would be great if you can share your solution.
